### PR TITLE
Feature/czar v2 8h price

### PR DESCRIPTION
**Pull request by jefferythewind**
## Summary

Adds volatility prediction support to AlloraMLWorkflow, integrates the CZAR directional loss function, and provides training scripts + deployed models for all mainnet-equivalent testnet topics.

### Core library (3 files, 248 lines) — please review carefully

- `allora_forge_builder_kit/workflow.py` — Adds `target_type` parameter (`"log_return"` | `"volatility"`) and `compute_volatility_target_polars()` for computing std of 1-min log returns over the prediction horizon. Used by volatility topics 79–82.
- `allora_forge_builder_kit/czar_loss.py` *(new)* — CZAR (Composite Zero-Agnostic Returns) loss with gradient/hessian for custom LightGBM training. Penalizes wrong-sign predictions, softens near-zero returns, normalizes by local volatility. Exported as `make_czar_objective`, `czar_loss`, `czar_gradient`, `czar_hessian`.
- `allora_forge_builder_kit/__init__.py` — Exports the new CZAR symbols.

### Tests (2 files, 146 lines)

- `test_volatility_target.py` — 8 tests covering volatility target computation, parameter validation, and target integrity.

### Docs/config (3 files)

- `README.md` — Updated topic reference tables with new testnet topics 83, 84. Added volatility workflow example.
- `AGENTS.md` — Fixed paths to reflect `testnet/` subfolder reorganization.
- `.gitignore` — Added `**/runs/` for training artifacts.

### Notebooks (38 files, ~14.5K lines) — example/illustration code, not mission-critical

Everything under `notebooks/` demonstrates how to use the core library. These are training scripts for practitioners, not part of the package. Key additions:

- `notebooks/testnet/topic_*/model_czar_v2.py` — CZAR V2 training scripts with joint grid search over loss params × LightGBM hyperparams × feature configurations
- `notebooks/testnet/topic_*/example.py` — Baseline MSE training scripts per topic
- Volatility model variants (topics 79–82): deep lookback, multiscale, GARCH-calibrated
- `notebooks/dashboard.sh` — Convenience script for worker status

### Reorganization

- All topic scripts moved from `notebooks/topic_*/` → `notebooks/testnet/topic_*/`
- Removed `notebooks/shared/` — deploy scripts stay at `notebooks/` top level
- `deploy_worker.py` and topic 69/77 walkthroughs unchanged

## Test suite

75 passed, 0 failed, 17 skipped.

---
## Summary by cubic
Adds volatility targets to `AlloraMLWorkflow` and a CZAR loss for directional training, plus full example coverage for 8h price, 8h/24h log‑return, and 15‑min volatility topics. Updates tests and docs, exports CZAR APIs, and adds a worker dashboard script.

- New Features
  - `AlloraMLWorkflow`: new `target_type` (`"log_return"` default, `"volatility"` = std of 1‑min log returns over horizon) and `compute_volatility_target_polars()` for topics 79–82.
  - New `czar_loss.py` with `make_czar_objective`, gradient, and hessian for custom LightGBM training; exported via `allora_forge_builder_kit/__init__.py`.
  - Added 8 tests for volatility target computation; `.gitignore` now ignores `**/runs/`.

- Docs and Examples
  - Updated `README.md`, `AGENTS.md`, and `skills/allora-model-builder/SKILL.md` with volatility workflow and refreshed topic tables.
  - Reorganized `notebooks/testnet/` with scripts for 38/41/42 (8h price, incl. CZAR models), 57/71/83/84 (8h log‑return), 61/62/63 (24h log‑return), and 79–82 (volatility, multiple variants).
  - Added `notebooks/dashboard.sh` for a quick worker dashboard.

<sup>Written for commit 4aee02b6585e30ff3909722f09ffea59799f66ec. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/allora-network/allora-forge-builder-kit/pull/30?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

File: skills/allora-model-builder/SKILL.md [hunk 1]
```diff
@@ -120,6 +120,64 @@ worker = AlloraWorker(
...
+
+def predict(nonce=None):
+    features = workflow.get_live_features("btcusd")
+    log_pred = model.predict(features[feature_cols].values.reshape(1, -1))[0]
+    vol = np.exp(log_pred) * bias_correction
+    return float(max(0.0, vol))
+```
+
+Volatility topics: 79 (BTC), 80 (ETH), 81 (XRP), 82 (SOL).
+See `notebooks/topic_79_btc_vol/topic_79_model_e_calibrated.py` for the full implementation.
```
[the remark this answers is not in the document]
Comment (jefferythewind) [reply skills/allora-model-builder/SKILL.md:180]: The SKILL.md path fix is up.
